tests/integration.rs

```diff
@@ -5992,26 +5992,62 @@ mod common_parallel {
     #[test]
     #[cfg(not(feature = "mshv"))]
     fn test_snapshot_restore_hotplug_virtiomem() {
-        _test_snapshot_restore(true);
+        _test_snapshot_restore(true, false);
     }
 
     #[test]
     fn test_snapshot_restore_basic() {
-        _test_snapshot_restore(false);
+        _test_snapshot_restore(false, false);
     }
 
-    fn _test_snapshot_restore(use_hotplug: bool) {
+    #[test]
+    fn test_snapshot_restore_with_fd() {
+        _test_snapshot_restore(false, true);
```

Comment:
I think it is worth adding some negative tests as well. Also unit tests for validate function to make sure error handling is proper as per the expectation.

Reply:
Added unit tests for error handlings. This will handle the negative scenarios. Let me know, if you think we need negative tests in integration suite as well.